diff --git a/.bazelrc b/.bazelrc
index 7e4a11b807..d1ad14c793 100644
--- a/.bazelrc
+++ b/.bazelrc
@@ -3,11 +3,11 @@ build --workspace_status_command=./hack/print-workspace-status.sh
 run --workspace_status_command=./hack/print-workspace-status.sh

 # https://github.com/kubernetes/test-infra/issues/13140
-build --incompatible_disable_deprecated_attr_params=false
+build #--incompatible_disable_deprecated_attr_params=false
 query --incompatible_disable_deprecated_attr_params=false

 # enable data race detection
-test --features=race --test_output=errors
+test --test_output=errors

 # only build tests when testing
 test --build_tests_only
diff --git a/load.bzl b/load.bzl
index 0a1fd340b8..3895899e2e 100644
--- a/load.bzl
+++ b/load.bzl
@@ -36,13 +36,17 @@ def repositories():
     http_archive(
         name = "io_bazel_rules_k8s",
         strip_prefix = "rules_k8s-0.5",
-        urls = ["https://github.com/bazelbuild/rules_k8s/archive/v0.5.tar.gz"],
+        patch_args = ["-p1"],
+        patches = ["@//:patch_k8s.diff"],
+        urls = ["https://github.com/bazelbuild/rules_k8s/archive/v0.5.tar.gz"],
         sha256 = "773aa45f2421a66c8aa651b8cecb8ea51db91799a405bd7b913d77052ac7261a",
     )

     # https://github.com/bazelbuild/rules_nodejs
     http_archive(
         name = "build_bazel_rules_nodejs",
+        patch_args = ["-p1"],
+        patches = ["@//:patch_nodejs.diff"],
         sha256 = "9abd649b74317c9c135f4810636aaa838d5bea4913bfa93a85c2f52a919fdaf3",
         urls = ["https://github.com/bazelbuild/rules_nodejs/releases/download/0.36.0/rules_nodejs-0.36.0.tar.gz"],
     )
diff --git a/prow/def.bzl b/prow/def.bzl
index 451e07b5c8..7c43c35675 100644
--- a/prow/def.bzl
+++ b/prow/def.bzl
@@ -35,7 +35,7 @@ def prow_image(
         name = app_name,
         base = base,
         embed = [":go_default_library"],
-        goarch = "amd64",
+        goarch = "s390x",
         goos = "linux",
         pure = "on",
         x_defs = {"k8s.io/test-infra/prow/version.Name": component},
diff --git a/containers.bzl b/containers.bzl
index 8927c767e9..a47cf62661 100644
--- a/containers.bzl
+++ b/containers.bzl
@@ -70,3 +70,10 @@ def repositories():
         repository = "k8s-testimages/launcher.gcr.io/google/bazel",
         tag = "v20200609-e7bfd25-test-infra",
     )
+    container_pull(
+        name = "alpine-base-s390x",
+        digest = "sha256:80e91290c38f8645d41cf790147dfde5806cc6a34d1f6ec093a0adff4265da6b",
+        registry = "docker.io",
+        repository = "alpine",
+        tag = "20200626",
+    )
diff --git a/prow/cmd/entrypoint/BUILD.bazel b/prow/cmd/entrypoint/BUILD.bazel
index 44d136f5dc..299af3730c 100644
--- a/prow/cmd/entrypoint/BUILD.bazel
+++ b/prow/cmd/entrypoint/BUILD.bazel
@@ -25,7 +25,7 @@ go_binary(

 prow_image(
     name = "image",
-    base = "@git-base//image",
+    base = "@alpine-base-s390x//image",
     component = NAME,
     symlinks = {"/entrypoint": "/app/prow/cmd/entrypoint/app.binary"},
     visibility = ["//visibility:public"],
