diff --git a/k8s/k8s.bzl b/k8s/k8s.bzl
index 35e54fd..cdbbae8 100644
--- a/k8s/k8s.bzl
+++ b/k8s/k8s.bzl
@@ -55,7 +55,8 @@ py_library(
     # Register the default kubectl toolchain targets for supported platforms
     # note these work with the autoconfigured toolchain
     native.register_toolchains(
-        "@io_bazel_rules_k8s//toolchains/kubectl:kubectl_linux_toolchain",
+        "@io_bazel_rules_k8s//toolchains/kubectl:kubectl_linux_amd64_toolchain",
+        "@io_bazel_rules_k8s//toolchains/kubectl:kubectl_linux_s390x_toolchain",
         "@io_bazel_rules_k8s//toolchains/kubectl:kubectl_osx_toolchain",
         "@io_bazel_rules_k8s//toolchains/kubectl:kubectl_windows_toolchain",
     )
diff --git a/toolchains/kubectl/BUILD b/toolchains/kubectl/BUILD
index 0f804e0..7cf3ed5 100644
--- a/toolchains/kubectl/BUILD
+++ b/toolchains/kubectl/BUILD
@@ -30,7 +30,7 @@ kubectl_toolchain(
 )

 toolchain(
-    name = "kubectl_linux_toolchain",
+    name = "kubectl_linux_amd64_toolchain",
     target_compatible_with = [
         "@bazel_tools//platforms:linux",
         "@bazel_tools//platforms:x86_64",
@@ -39,6 +39,16 @@ toolchain(
     toolchain_type = ":toolchain_type",
 )

+toolchain(
+    name = "kubectl_linux_s390x_toolchain",
+    target_compatible_with = [
+        "@bazel_tools//platforms:linux",
+        "@bazel_tools//platforms:s390x",
+    ],
+    toolchain = "@k8s_config//:toolchain",
+    toolchain_type = ":toolchain_type",
+)
+
 toolchain(
     name = "kubectl_osx_toolchain",
     target_compatible_with = [
